rust/datafusion/src/logical_plan/mod.rs

```diff
@@ -1089,6 +1079,15 @@ pub fn can_coerce_from(type_into: &DataType, type_from: &DataType) -> bool {
     }
 }
 
+/// A registry knows how to build logical expressions out of user-defined function' names
+pub trait Registry {
```

Comment:
I wonder if renaming this to `FunctionRegistry` might make it a little more specific, without being overly verbose. I don't feel strongly about this change

```suggestion
pub trait FunctionRegistry {
```

Reply:
[Explicit is better than implicit](https://www.python.org/dev/peps/pep-0020/#id2) (done)